python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py

```diff
@@ -1279,8 +1267,54 @@ def resolve_assetChecksOrError(
         limit=None,
         pipeline: Optional[GraphenePipelineSelector] = None,
     ) -> AssetChecksOrErrorUnion:
-        return self._asset_checks_loader.get_checks_for_asset(
-            self._asset_node_snap.asset_key, limit, pipeline
+        remote_check_nodes = graphene_info.context.asset_graph.get_checks_for_asset(
+            self._asset_node_snap.asset_key
+        )
+        if not remote_check_nodes:
+            return GrapheneAssetChecks(checks=[])
+
+        asset_check_support = graphene_info.context.instance.get_asset_check_support()
+        if asset_check_support == AssetCheckInstanceSupport.NEEDS_MIGRATION:
+            return GrapheneAssetCheckNeedsMigrationError(
+                message="Asset checks require an instance migration. Run `dagster instance migrate`."
+            )
+        elif asset_check_support == AssetCheckInstanceSupport.NEEDS_AGENT_UPGRADE:
+            return GrapheneAssetCheckNeedsAgentUpgradeError(
+                "Asset checks require an agent upgrade to 1.5.0 or greater."
+            )
+        else:
+            check.invariant(
+                asset_check_support == AssetCheckInstanceSupport.SUPPORTED,
+                f"Unexpected asset check support status {asset_check_support}",
+            )
+
+        library_versions = graphene_info.context.get_dagster_library_versions(
+            self._repository_handle.location_name
+        )
+        code_location_version = (library_versions or {}).get("dagster")
+        if code_location_version and version.parse(code_location_version) < version.parse("1.5"):
+            return GrapheneAssetCheckNeedsUserCodeUpgrade(
+                message=(
+                    "Asset checks require dagster>=1.5. Upgrade your dagster"
+                    " version for this code location."
+                )
+            )
+        if pipeline:
+            remote_check_nodes = [
+                node
+                for node in remote_check_nodes
+                if node.handle.location_name == pipeline.repositoryLocationName
+                and node.handle.repository_name == pipeline.repositoryName
+                and pipeline.pipelineName in node.asset_check.job_names
+            ]
+
+        if limit:
+            remote_check_nodes = remote_check_nodes[:limit]
```

Comment:
these args seemed like it would be worse when i was building (which is why i dropped it) but actually v straightforward

Reply:
hm i guess it doesn't really give any perf benefit though, huh

I guess maybe if data is then being fetched on the subsequent objects